### DISCUSSION TITLE
Inter-Module Communication Between City and ItemListing Modules in Oqtane

### DISCUSSION THREAD

**Discussion by techravi007** [oqtane/oqtane.framework#4794] [Q&A] [answered]
Hi everyone,

I'm working on a project with two modules in Oqtane: a City module that displays a dropdown list of cities and a ItemListing module that needs to load products based on the selected city.

I would like to implement inter-module communication so that when a user selects a city in the City module, the ItemListing module automatically updates to show products for that city.

I've considered using an Event Aggregator pattern for this, but I’m looking for advice on the best practices for implementing this. Specifically:

How should I structure the event for the selected city?
What is the best way to publish and subscribe to this event between the two modules?
Are there any pitfalls or considerations I should be aware of while implementing this?
Any guidance or examples would be greatly appreciated!

PS: I have checked this thread-https://github.com/oqtane/oqtane.framework/discussions/4592
but seems the ask is different in this thread. 

Thanks in advance!
🙏

**Comment by leigh-pointer** [comment, marked as answer]
Hi @techravi007 
This will get you started and what you want https://youtu.be/Yvp_PLvJqqk?t=1636

It shows how to use the INotifyPropertyChangeEvent implement in Oqtane

**Reply by techravi007** [reply]
Hi @leigh-pointer 
Thanks for the pointer. `INotifyPropertyChangeEvent ` did the job.

**Reply by leigh-pointer** [reply]
@techravi007 i'm pleased i could help.
Can you mark this as answered, please.
